lib/awful/titlebar.lua

```diff
@@ -754,14 +811,20 @@ end
 -- @constructorfct awful.titlebar.widget.button
 function titlebar.widget.button(c, name, selector, action)
     local ret = imagebox()
-
     if titlebar.enable_tooltip then
-        ret._private.tooltip = atooltip({ objects = {ret}, delay_show = 1 })
-        ret._private.tooltip:set_text(name)
+        ret._private.tooltip = atooltip({
+            objects = {ret},
+            delay_show = beautiful["titlebar_tooltip_delay_show"] or 1.0,
+            margins_leftright = beautiful["titlebar_tooltip_margins_leftright"],
+            margins_topbottom = beautiful["titlebar_tooltip_margins_topbottom"],
+            timeout = beautiful["titlebar_tooltip_timeout"],
+            align = beautiful["titlebar_tooltip_align"]
+        })
     end
 
     local function update()
         local img = selector(c)
+        local tooltip_text = name
```

Comment:
I think this variable declaration here can be moved to the assignation on line 850 since it's not used before.

Reply:
Moved to the inner if-condition section.